src/storage/src/types/sources.rs

```diff
@@ -1489,6 +1506,67 @@ impl RustType<ProtoPubNubSourceConnection> for PubNubSourceConnection {
     }
 }
 
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct LoadGeneratorSourceConnection {
+    pub generator: Generator,
+}
+
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
```

Comment:
I don’t know this code base well enough to give a meaningful review. My one ask, could you add some doc comments about how to add new generator types? It would be great to be in a position where myself and the devex team can come in independently and add new types as needed.